aws/cloudformation/bootstrap_chef_stack.sh.erb

```diff
@@ -24,11 +24,61 @@ INSTANCE_ID=$(curl -s http://169.254.169.254/latest/meta-data/instance-id)
 NODE_NAME=<%=node_name%>
 RESOURCE_ID=<%=resource_id%>
 
+<% if environment == 'adhoc' -%>
+# Redirect copy of stdout/stderr to a log file for later auditing.
+LOG=<%=log_name%>
+exec >> >(tee -i $LOG)
+exec 2>&1
+
+# Temp dir that cleans up after itself.
+MYTMPDIR=$(mktemp -d)
+trap 'rm -rf "$MYTMPDIR"' EXIT
+
+# Install AWS CloudWatch Logs
+# Ref: http://docs.aws.amazon.com/AmazonCloudWatch/latest/DeveloperGuide/EC2NewInstanceCWL.html
+AWSLOGS_AGENT_URL=https://s3.amazonaws.com/aws-cloudwatch/downloads/latest/awslogs-agent-setup.py
+AWSLOGS_CONF=$MYTMPDIR/awslogs.conf
+cat <<CONF > $AWSLOGS_CONF
+[general]
+region = $REGION
+[$LOG]
+log_group_name = $STACK
+log_stream_name = $LOG
+file = $LOG
+CONF
+curl -s $AWSLOGS_AGENT_URL | python - -n -r $REGION -c $AWSLOGS_CONF > /dev/null
+aws logs put-retention-policy \
+  --log-group-name $STACK \
+  --retention-in-days 1 \
+  --region $REGION
+<% end -%>
+<% unless @load_balancer -%>
+# Fetch acmesmith-provisioned SSL certificate and key from S3
+aws configure set s3.signature_version s3v4
+S3_PATH=s3://$S3_BUCKET/ssl/certs/$SUBDOMAIN
+CURRENT=$S3_PATH/$(aws s3 cp $S3_PATH/current -)
+CERT=$(aws s3 cp $CURRENT/cert.pem -)
+CHAIN=$(aws s3 cp $CURRENT/chain.pem -)
+KEY=$(aws s3 cp $CURRENT/key.pem -)
+<% end -%>
 # Customize Chef configuration.
 FIRST_BOOT=/etc/chef/first-boot.json
 mkdir -p $(dirname $FIRST_BOOT)
 cat <<JSON > $FIRST_BOOT
 {
+<% unless @load_balancer -%>
+  "cdo-nginx": {
+    "ssl_cert": {
+      "content": "$CERT"
+    },
+    "ssl_chain": {
+      "content": "$CHAIN"
+    },
+    "ssl_key": {
+      "content": "$KEY"
+    }
+  },
+<% end -%>
```

Comment:
These above additions make the bootstrap script more or less equivalent to the `standalone` version for adhoc instance creation- the new additions are contained within `if environment == 'adhoc'` or `unless @load_balancer` conditionals so there's no change to non-adhoc environments.

Reply:
The adhoc instance I provisioned using these changes presented a self-signed certificate with `cdn-code.org` as the common name.